tools/telemetry-generator/src/handlers/appInsightsExecutionTimeTestHandler.ts

```diff
@@ -14,12 +14,17 @@ module.exports = function handler(fileData, telemetryClient: TelemetryClient): v
 	for (const testData of fileData.benchmarks) {
 		const arithmeticMeanMetricName = `${fileData.suiteName}_${testData.benchmarkName}_arithmeticMean`;
 		try {
+			const value = testData.customData["Period (ns/op)"];
+			if (isNaN(value)) {
```

Comment:
Do we expect non-number data to come in via `customData`? Do our types allow that?

If we do support non-number data, then we shouldn't call `console.error`.

If we don't support non-number data, then I wonder if we should just throw an error here?

Reply:
Types don't quite apply here, I think. The handlers in this package process json files produced by other processes, so we can't really guarantee type safety.

The check needs to be different though, If the value in `customData["Period (ns/op)"]` is a string, `isNaN` will return false but that's still a bad value. So probably something like `if (Number.isNaN(Number.parseFloat(value)))`.

Reply:
> If we don't support non-number data, then I wonder if we should just throw an error here?

I was debating the same thing. For the app insights handlers I think it doesn't make a difference, the try-catch will handle it and we'll just continue trying to emit the next metric. For the non-app-insight handlers, where we send a single telemetry event with all the values... I think I do lean towards just throwing and terminating the process instead of trying to submit any other potentially valid values. Clear sign of "something that we absolutely do not expect happened so I had to terminate" instead of just a log message that we will not notice until someone is actually looking for that data and can't find it in Kusto.

Yeah, for the non-app-insights handlers, let's just `throw new Error(<message>)` @chentong7 .

Reply:
Updated